avro2tf/src/main/scala/com/linkedin/avro2tf/jobs/PrepRankingData.scala

```diff
@@ -122,13 +121,15 @@ object PrepRankingData {
             skipPadding = params.skipPadding)(col(it)))
     }
 
+    val twoDimSparseFeatureSchema = generateSchemaForMultiDimSparseFeature(2)
```

Comment:
I'd suggest changing this to `val twoDimSparseVectorArray = multiDimSparseVectorArray(generateSchemaForMultiDimSparseFeature(2))`.

Then, we can reuse this udf on line 132 instead of registering a new udf for each sparse-vector document feature.

Reply:
Sounds good. Fixed